src/attrs_strict/_type_validation.py

```diff
@@ -248,6 +256,29 @@ def _handle_union(attribute, value, expected_type):
     raise UnionError(value, attribute.name, expected_type)
 
 
+def _handle_literal(attribute, value, expected_type):
+    flattened_literals = _flatten_literals(expected_type)
+
+    if not any(
+        value == literal and type(value) == type(literal)
```

Comment:
Does this work for recursive types such as list of dictionaries ?

Reply:
Legal parameters for Literal at type check time
* Literal may be parameterized with literal ints, byte and unicode strings, bools, Enum values and None.
* Literal may also be parameterized by other literal types, or type aliases to other literal types.

Anything else inside Literal is invalid at type check time.

https://www.python.org/dev/peps/pep-0586/#id23

Reply:
Although at type check time only what I mentioned is valid but anything else would still be a valid python code. Current implementation will break client code when something like `Literal[Dict]` comes.

TODO: Need to handle that case by ignoring type check.

Reply:
It's going to be hard to achieve corectness in this case, and we can restrict ourselves to primitive types mentioned above. However something like this, is still valid:
```python
>>> x = typing.Literal[[1, 2, 3]]
>>> x.__args__
([1, 2, 3],)
```
Making it work for this is going to be **slightly** tricky.